app/test/grades/grades_service_test.dart

```diff
@@ -40,5 +42,75 @@ void main() {
         isNull,
       );
     });
+
+    test('deleteSubject removes subject and grades from all terms', () {
```

Comment:
If you're testing for terms and grades (plural) then you should actually test it with several terms and grades IMO. Maybe 2 terms, one with 2 grades from the subject ot be deleted and one with 2 grades, 1 to be deleted one from another subject. In the end make sure the grades that should be deleted are deleted and the grade that should not be deleted is not deleted.

(One could argue that one should separate this into two tests (deleting everything it should, not deleting anything it should not) but that's kleinscheißerei, I don't care :D)